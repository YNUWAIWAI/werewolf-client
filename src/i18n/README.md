# i18n

## Message Syntax
https://formatjs.io/guides/message-syntax/

## id

`()`内の値は実行時に決定されることを示す．
