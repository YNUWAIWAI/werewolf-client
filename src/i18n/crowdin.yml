project_identifier_env: CROWDIN_PROJECT_ID
api_key_env: CROWDIN_API_KEY
base_path : "./"
preserve_hierarchy: true

files:
  -
    source: '*/en.json'
    translation: '%original_path%/%locale%.json'
    languages_mapping:
      locale:
        'af': 'af'
        'ar': 'ar'
        'bs-BA': 'bs-BA'
        'ca': 'ca'
        'cs': 'cs'
        'da': 'da'
        'de': 'de'
        'el': 'el'
        'es-ES': 'es-ES'
        'fa': 'fa-IR'
        'fi': 'fi'
        'fr': 'fr'
        'he': 'he'
        'hu': 'hu'
        'id': 'id-ID'
        'it': 'it'
        'ja': 'ja'
        'ko': 'ko'
        'mr': 'mr-IN'
        'nl': 'nl'
        'no': 'no-NO'
        'pl': 'pl'
        'pt-BR': 'pt-BR'
        'pt-PT': 'pt-PT'
        'ro': 'ro'
        'ru': 'ru'
        'sk': 'sk-SK'
        'sr': 'sr'
        'sv-SE': 'sv-SE'
        'tr': 'tr'
        'uk': 'uk'
        'vi': 'vi'
        'zh-CN': 'zh-CN'
        'zh-TW': 'zh-TW'
