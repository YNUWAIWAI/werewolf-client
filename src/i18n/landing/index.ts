import enMessages from './en'
import frMessages from './fr'
import itMessages from './it'
import jaMessages from './ja'

export const enum Language {
  en = 'en',
  fr = 'fr',
  it = 'it',
  ja = 'ja'
}
interface Messages {
  [id: string]: string
}

export const getMessages = (locale: Language): Messages => {
  switch (locale) {
    case 'en':
      return enMessages
    case 'fr':
      return frMessages
    case 'it':
      return itMessages
    case 'ja':
      return jaMessages
    default:
      throw Error(`${locale} is unsuported`)
  }
}
