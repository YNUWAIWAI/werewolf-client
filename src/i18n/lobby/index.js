// @flow
import type {Language} from 'village'
import en from './en'
import fr from './fr'
import it from './it'
import ja from './ja'

export const getMessages = (locale: Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'fr':
      return fr
    case 'it':
      return it
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
