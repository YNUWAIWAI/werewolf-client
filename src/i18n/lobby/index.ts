/// <reference types="lobby" />
import en from './en'
import fr from './fr'
import it from './it'
import ja from './ja'

export const getMessages = (locale: lobby.Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'fr':
      return fr
    case 'it':
      return it
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
