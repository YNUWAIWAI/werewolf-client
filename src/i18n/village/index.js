// @flow
import type {Language} from 'village'
import en from './en'
import ja from './ja'

export const getMessages = (locale: Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
