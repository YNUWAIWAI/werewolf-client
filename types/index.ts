import * as lobby from './lobby'
import * as village from './village'

export {
  lobby,
  village
}
