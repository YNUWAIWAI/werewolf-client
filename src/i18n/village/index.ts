import * as village from '../../../types/village'
import en from './en.json'
import fr from './fr.json'
import it from './it.json'
import ja from './ja.json'

export const getMessages = (locale: village.Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'fr':
      return fr
    case 'it':
      return it
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
