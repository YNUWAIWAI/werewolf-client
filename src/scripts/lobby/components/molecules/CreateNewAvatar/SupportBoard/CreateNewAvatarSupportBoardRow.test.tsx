import * as React from 'react'
import {CreateNewAvatarSupportBoardRow} from './CreateNewAvatarSupportBoardRow'
import {getMessages} from '../../../../i18n'
import {initRenderer} from '../../../../tools'
import {lobby} from '../../../../types'

const {mountWithIntl} = initRenderer(lobby.Language.en, getMessages(lobby.Language.en))

test('render', () => {
  const wrapper = mountWithIntl(
    <CreateNewAvatarSupportBoardRow
      className="className"
      numberOfPlayers={15}
      support={{
        [lobby.Member.A]: false,
        [lobby.Member.B]: false,
        [lobby.Member.C]: false
      }}
    />
  )

  expect(wrapper.html()).toMatchSnapshot()
})
