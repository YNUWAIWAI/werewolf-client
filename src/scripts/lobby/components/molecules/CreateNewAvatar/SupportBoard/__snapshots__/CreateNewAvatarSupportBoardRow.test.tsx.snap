// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `"<div class=\\"className\\">15</div><div class=\\"className\\"></div><div class=\\"className\\"></div><div class=\\"className\\"></div>"`;
